### PR TITLE
Add RuneScape manifest

### PR DESCRIPTION
**Pull request by Limero**

### CODE REVIEW

File: com.jagex.RuneScape.json [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        "/lib/libXmuu*",
+        "/lib/libXmu.so",
+        "/lib/libGLEW.so",
+        "/lib/libpng.*"
+    ],
+    "modules": [
+        {
+            "name": "libpng",
+            "config-opts": ["--disable-static"],
+            "cleanup": ["/bin", "/include", "/lib/debug", "/lib/pkgconfig", "/share"],
```
Comment (TingPing) [inline com.jagex.RuneScape.json]: Don't cleanup the debug directories for any of the deps.

File: com.jagex.RuneScape.json [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                    "type": "archive",
+                    "url": "https://xorg.freedesktop.org/releases/individual/lib/libXmu-1.1.2.tar.bz2",
+                    "sha256": "756edc7c383254eef8b4e1b733c3bf1dc061b523c9f9833ac7058378b8349d0b"
+                }
+            ]
+        },
+        {
+            /* No textures will be displayed without this */
+            "name": "s2tc",
+            "config-opts": ["--disable-tools"],
```
Comment (TingPing) [inline com.jagex.RuneScape.json]: `--disable-static`

File: com.jagex.RuneScape.json [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                    "type": "script",
+                    "dest-filename": "apply_extra",
+                    "commands": [
+                        "ar x runescape.deb",
+                        "tar xf data.tar.xz",
+                        "rm -f runescape.deb control.tar.gz data.tar.xz debian-binary _gpgbuilder",
+                        "mv usr/share/games/runescape-launcher/runescape .",
+
+                        "mkdir -p export/share/applications",
+                        "sed -i 's|/usr/bin/runescape-launcher %u|runescape|g' usr/share/applications/runescape-launcher.desktop",
```
Comment (TingPing) [inline com.jagex.RuneScape.json]: This is removing `%u`, what are urls used for, probably want to keep that support?

File: com.jagex.RuneScape.json [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                        "sed s/Icon=runescape/Icon=com.jagex.RuneScape/ usr/share/applications/runescape-launcher.desktop > export/share/applications/com.jagex.RuneScape.desktop",
+
+                        "mv usr/share/icons export/share/",
+                        "for d in export/share/icons/hicolor/*; do mv $d/apps/runescape.png $d/apps/com.jagex.RuneScape.png; done;",
```
Comment (TingPing) [inline com.jagex.RuneScape.json]: Maybe it is cleaner to use the `rename-icon` flatpak option and let it handle this? Removes the `sed` call above too.

File: com.jagex.RuneScape.json [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        "cflags": "-O2 -g",
+        "cxxflags": "-O2 -g",
+        "env": {
+            "V": "1"
+        }
+    },
+    "cleanup": [
+        "*.a",
+        "*.la",
+        "/lib/libXmuu*",
```
Comment (TingPing) [inline com.jagex.RuneScape.json]: Maybe there is a configure opt to not build this lib at all?

File: com.jagex.RuneScape.json [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        "/lib/libXmu.so",
```
Comment (TingPing) [inline com.jagex.RuneScape.json]: These are just symlinks right? Can just leave em be rather than complicating the manifest.

**Comment by TingPing** [comment]
You will want a `flathub.json` file like this one to mark it only buildable for specific arches: https://github.com/flathub/com.spotify.Client/blob/master/flathub.json

File: com.jagex.RuneScape.json (hunk 5 above)
Comment (Limero) [inline com.jagex.RuneScape.json]: I could not find anything on this.

File: com.jagex.RuneScape.json (hunk 6 above)
Comment (Limero) [inline com.jagex.RuneScape.json]: Those are are separate versions not used by the application. There are also symlinks and they are left alone.

**Comment by Limero** [comment]
I've also decided to pass the arguments over to the launch script in the .deb and let runescape.sh just be a pass-through to avoid having the same script in two places.

**Comment by TingPing** [comment]
Test build: https://flathub.org/builds/#/builders/5/builds/443

**Comment by TingPing** [comment]
Build failure, forgot to install the appdata file.

**Comment by Barthalion** [comment]
Build [succeeded](https://flathub.org/builds/#/builders/5/builds/465).

**Comment by TingPing** [comment]
Repository has been created: https://github.com/flathub/com.jagex.RuneScape

**Pull request closed without merging**